pkg/vault/test/foundry/BatchRouterERC4626Pool.t.sol

```diff
@@ -0,0 +1,409 @@
+
+        uint256 afterDAIBalance = dai.balanceOf(sender);
+        assertEq(beforeDAIBalance - afterDAIBalance, 0, "DAI balance should be the same");
+
+        (uint256 afterWaUSDCBufferBalanceUnderlying, uint256 afterWaUSDCBufferBalanceWrapped) = vault.getBufferBalance(
+            waUSDC
+        );
+        assertEq(
+            beforeWaUSDCBufferBalanceWrapped - afterWaUSDCBufferBalanceWrapped,
+            0,
+            "waUSDC wrapped buffer balance should be the same"
+        );
+        assertEq(
+            afterWaUSDCBufferBalanceUnderlying - beforeWaUSDCBufferBalanceUnderlying,
+            0,
+            "waUSDC underlying buffer balance should be the same"
+        );
+
+        (uint256 afterWaDAIBufferBalanceUnderlying, uint256 afterWaDAIBufferBalanceWrapped) = vault.getBufferBalance(
+            waDAI
+        );
+        assertEq(
+            beforeWaDAIBufferBalanceWrapped - afterWaDAIBufferBalanceWrapped,
+            0,
+            "waDAI wrapped buffer balance should be the same"
+        );
+        assertEq(
+            afterWaDAIBufferBalanceUnderlying - beforeWaDAIBufferBalanceUnderlying,
+            0,
+            "waDAI underlying buffer balance should be the same"
+        );
+    }
+
+    function testAddLiquidityUnbalancedToERC4626Pool() public {
+        uint256[] memory exactUnderlyingAmountsIn = [operationAmount, operationAmount].toMemoryArray();
+
+        _prankStaticCall();
+        uint256 expectBPTOut = router.queryAddLiquidityUnbalanced(erc4626Pool, exactUnderlyingAmountsIn, new bytes(0));
+
+        uint256 beforeUSDCBalance = usdc.balanceOf(alice);
+        uint256 beforeDAIBalance = dai.balanceOf(alice);
+        (uint256 beforeWaUSDCBufferBalanceUnderlying, uint256 beforeWaUSDCBufferBalanceWrapped) = vault
+            .getBufferBalance(waUSDC);
+        (uint256 beforeWaDAIBufferBalanceUnderlying, uint256 beforeWaDAIBufferBalanceWrapped) = vault.getBufferBalance(
+            waDAI
+        );
+
+        vm.prank(alice);
+        uint256 bptOut = batchRouter.addLiquidityUnbalancedToERC4626Pool(
+            erc4626Pool,
+            exactUnderlyingAmountsIn,
+            1,
+            false,
+            new bytes(0)
+        );
+
+        {
+            uint256 afterUSDCBalance = usdc.balanceOf(alice);
+            assertEq(beforeUSDCBalance - afterUSDCBalance, operationAmount, "USDC balance should decrease");
+        }
+        {
+            uint256 afterDAIBalance = dai.balanceOf(alice);
+            assertEq(beforeDAIBalance - afterDAIBalance, operationAmount, "DAI balance should decrease");
+        }
+        {
+            (uint256 afterWaUSDCBufferBalanceUnderlying, uint256 afterWaUSDCBufferBalanceWrapped) = vault
+                .getBufferBalance(waUSDC);
+            assertEq(
+                beforeWaUSDCBufferBalanceWrapped - afterWaUSDCBufferBalanceWrapped,
+                operationAmount,
+                "waUSDC wrapped buffer balance should decrease"
+            );
```

Comment:
```suggestion
            assertEq(
                afterWaUSDCBufferBalanceWrapped,
                beforeWaUSDCBufferBalanceWrapped - operationAmount,
                "waUSDC wrapped buffer balance should decrease"
            );
```

Super minor, but maybe writing it like this is easier to read with this error message.